src/video_core/renderer_vulkan/vk_instance.h

```diff
@@ -0,0 +1,287 @@
+// Copyright 2022 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <span>
+
+#include "video_core/rasterizer_cache/pixel_format.h"
+#include "video_core/regs_pipeline.h"
+#include "video_core/renderer_vulkan/vk_platform.h"
+
+namespace Frontend {
+class EmuWindow;
+}
+
+namespace VideoCore {
+enum class CustomPixelFormat : u32;
+}
+
+VK_DEFINE_HANDLE(VmaAllocator)
+
+namespace Vulkan {
+
+struct FormatTraits {
+    bool transfer_support = false;
+    bool blit_support = false;
+    bool attachment_support = false;
+    bool storage_support = false;
+    bool needs_conversion = false;
+    bool needs_emulation = false;
+    vk::ImageUsageFlags usage{};
+    vk::ImageAspectFlags aspect;
+    vk::Format native = vk::Format::eUndefined;
+};
+
+class Instance {
+public:
+    explicit Instance(bool validation = false, bool dump_command_buffers = false);
+    explicit Instance(Frontend::EmuWindow& window, u32 physical_device_index);
+    ~Instance();
+
+    /// Returns the FormatTraits struct for the provided pixel format
+    const FormatTraits& GetTraits(VideoCore::PixelFormat pixel_format) const;
+    const FormatTraits& GetTraits(VideoCore::CustomPixelFormat pixel_format) const;
+
+    /// Returns the FormatTraits struct for the provided attribute format and count
+    const FormatTraits& GetTraits(Pica::PipelineRegs::VertexAttributeFormat format,
+                                  u32 count) const;
+
+    /// Returns the Vulkan instance
+    vk::Instance GetInstance() const {
+        return *instance;
+    }
+
+    /// Returns the current physical device
+    vk::PhysicalDevice GetPhysicalDevice() const {
+        return physical_device;
+    }
+
+    /// Returns the Vulkan device
+    vk::Device GetDevice() const {
+        return *device;
+    }
+
+    /// Returns the VMA allocator handle
+    VmaAllocator GetAllocator() const {
+        return allocator;
+    }
+
+    /// Returns a list of the available physical devices
+    std::span<const vk::PhysicalDevice> GetPhysicalDevices() const {
+        return physical_devices;
+    }
+
+    /// Retrieve queue information
+    u32 GetGraphicsQueueFamilyIndex() const {
+        return queue_family_index;
+    }
+
+    u32 GetPresentQueueFamilyIndex() const {
+        return queue_family_index;
+    }
+
+    vk::Queue GetGraphicsQueue() const {
+        return graphics_queue;
+    }
+
+    vk::Queue GetPresentQueue() const {
+        return present_queue;
+    }
+
+    /// Returns true if logic operations need shader emulation
+    bool NeedsLogicOpEmulation() const {
+        return !features.logicOp;
+    }
+
+    bool UseGeometryShaders() const {
+#ifndef __ANDROID__
```

Comment:
```suggestion
#ifdef __ANDROID__
```

Let's handle the positive case first